### PR TITLE
Fix debug messages

### LINKED ISSUE #335: tldr is showing a debug message when a page was not found
When I run tldr on a command which could not be found, I get the following output:
```
✔ Page not found. Updating cache...
✔ Creating index...
Error: Page not found.
If you want to contribute it, feel free to send a pull request to: https://github.com/tldr-pages/tldr
    at /usr/lib/node_modules/tldr/lib/tldr.js:147:17
```
Does the last line have any purpose? It's always the same and not really useful.
(tldr version 3.3.7)

### PR DESCRIPTION
**Pull request by marchersimon**
Closes #335

### CODE REVIEW

File: bin/tldr [hunk 1]
```diff
@@ -119,7 +120,11 @@ if (p === null) {
   process.exitCode = 1;
 } else {
   p.catch((err) => {
-    console.error(err);
+    if(err instanceof TldrError) {
```
Comment (bl-ue) [inline bin/tldr:123]:
```suggestion
    if (err instanceof TldrError) {
```

**Review by bl-ue — CHANGES_REQUESTED** [review]

**Review by bl-ue — APPROVED** [review]

**Comment by vladimyr** [comment]
@marchersimon Please apply these changes: https://github.com/vladimyr/tldr-node-client/commit/51d280c

The easiest way to do that is:
```sh
# reset your branch to master
git reset --hard master
# download patch from Github
curl -sSL https://github.com/vladimyr/tldr-node-client/commit/51d280c.patch -o debug-message.patch
# create commit from patch
git am debug-message.patch
```

Then you'll need to force push your branch but don't forget to use `--force-with-lease` flag :warning: 
```sh
git push --force-with-lease
```

File: lib/errors.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+module.exports = {
+  TldrError,
+  EmptyCacheError,
+  MissingPageError
+};
+
+function trim(strings, ...values) {
+  let output = values.reduce((acc, value, i) => {
+    return acc + strings[i] + value;
+  }, '') + strings[values.length];
```
Comment (bl-ue) [inline lib/errors.js:47]: @vladimyr is there a reason you didn't do this?
```suggestion
  let output = values.reduce((acc, value, i) => acc + strings[i] + value, '') + strings[values.length];
```

File: lib/errors.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  }
+
+  static isTldrError(err) {
+    return err instanceof this;
+  }
+}
+
+class EmptyCacheError extends TldrError {
+  constructor() {
+    super(trim`
```
Comment (bl-ue) [inline lib/errors.js:16]: Cool trick! ⚡

File: lib/errors.js (hunk 2 above)
Comment (vladimyr) [reply lib/errors.js:47]: Unfortunately, ESLint wouldn't let me do it 🙁
https://github.com/tldr-pages/tldr-node-client/blob/7e49dda42a11dfc07b5547b1bc9a20bc334b8792/.eslintrc.json#L9

Comment (bl-ue) [reply lib/errors.js:47]: @MasterOdin / @agnivade do you have any preferences regarding this style?  If you don't I'd like to relax this rule.

Comment (vladimyr) [reply lib/errors.js:47]: My guess is it will effectively double down the LOC amount...

**Comment by marchersimon** [comment]
Can this be merged?

File: lib/errors.js (hunk 2 above)
Comment (bl-ue) [reply lib/errors.js:47]: Let's wait for a different time.

**Review by bl-ue — APPROVED** [review]
It works!

**Review by owenvoke — APPROVED** [review]

File: lib/errors.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
+'use strict';
+
+class TldrError extends Error {
+  constructor(message) {
+    super(message);
+    this.name = 'TldrError';
+  }
+
+  static isTldrError(err) {
```
Comment (agnivade) [inline lib/errors.js:9]: Can we make this a global function rather than a static method?

File: lib/errors.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+class EmptyCacheError extends TldrError {
+  constructor() {
+    super(trim`
+Local cache is empty
+Please run tldr --update
+    `);
+    this.name = 'EmptyCacheError';
+    // eslint-disable-next-line no-magic-numbers
+    this.code = 2;
```
Comment (agnivade) [inline lib/errors.js:22]: What is the purpose of having error codes in the class object?

File: lib/errors.js [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+  }
+}
+
+module.exports = {
+  TldrError,
+  EmptyCacheError,
+  MissingPageError
+};
+
+function trim(strings, ...values) {
```
Comment (agnivade) [inline lib/errors.js:44]: What is the purpose of this? Is it just to remove the newlines in the original string?

File: lib/errors.js (hunk 4 above)
Comment (vladimyr) [reply lib/errors.js:9]: Sure but why? It'd just pollute the global namespace without any good reason?

File: lib/errors.js (hunk 5 above)
Comment (vladimyr) [reply lib/errors.js:22]: To reuse it as exit code as spec mandates different exit codes for different fatal errors.

File: lib/errors.js (hunk 6 above)
Comment (vladimyr) [reply lib/errors.js:44]: Syntactical sugar to replace the following form:
```js
    super(`
Page not found.
If you want to contribute it, feel free to send a pull request to: ${repo}
    `.trim());
```
with slightly more readable IMHO:
```js
    super(trim`
Page not found.
If you want to contribute it, feel free to send a pull request to: ${repo}
    `);
```

File: lib/errors.js (hunk 4 above)
Comment (agnivade) [reply lib/errors.js:9]: Ok.

**Review by agnivade — APPROVED** [review]

**Pull request merged**